tests/send_and_receive.rs

```diff
@@ -0,0 +1,164 @@
+// Copyright 2016 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under (1) the MaidSafe.net Commercial License,
+// version 1.0 or later, or (2) The General Public License (GPL), version 3, depending on which
+// licence you accepted on initial access to the Software (the "Licences").
+//
+// By contributing code to the SAFE Network Software, or to this project generally, you agree to be
+// bound by the terms of the MaidSafe Contributor Agreement, version 1.0.  This, along with the
+// Licenses can be found in the root directory of this project at LICENSE, COPYING and CONTRIBUTOR.
+//
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.
+//
+// Please review the Licences for the specific language governing permissions and limitations
+// relating to use of the SAFE Network Software.
+
+extern crate crust;
+#[macro_use]
+extern crate maidsafe_utilities;
+
+use crust::{CrustEventSender, Event, Service};
+use maidsafe_utilities::event_sender::{MaidSafeEventCategory, MaidSafeObserver};
+use std::sync::mpsc::{self, Receiver};
+use std::thread::{self, JoinHandle};
+
+const BEACON_PORT: u16 = 44444;
```

Comment:
The problem is that we cannot just bind a socket to steal the port as it is UDP. So there is no way knowing somebody else isn't using the port already. As it is for service discovery, it should be less problematic. The test could still fail if they discover nodes from a different node also running service discovery.

The only option to really fix the issue here would be to use rendezvous connections and manually connect each node, but it is very unlikely we'll step upon the fault case. Plus, this "safety" would add to test robustness, not to crust robustness, so it is not that critical. And robustness this time means avoiding false negatives, so it is even more justifiable to accept this simpler code.